jetbrains.jdk: Fix darwin build

felixscheinost opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Similarly to how OpenJDK is a source build on Linux but using binaries from Azul the source build of JetbrainsRuntime (OpenJDK fork) is currently broken on Darwin.

I think it makes sense to use binaries for Darwin here as well.

I also updated from 11_0_13-b1751.25 to 11_0_14_1-b2043.45 as we currently have IDE version 2021.3 for which https://github.com/JetBrains/JetBrainsRuntime/blob/master/README.md recommends this version.

I tested using `jetbrains.idea-community` which seems to work.

I am not sure yet how the directory structure of the resulting package should be.

Currently it looks like this 

```
result/
└── Contents
    ├── CodeResources
    ├── Frameworks
    │   ├── Chromium\ Embedded\ Framework.framework
    │   ├── JavaNativeFoundation.framework
    │   ├── jcef\ Helper\ (GPU).app
    │   ├── jcef\ Helper\ (Plugin).app
    │   ├── jcef\ Helper\ (Renderer).app
    │   └── jcef\ Helper.app
    ├── Home
    │   ├── bin
    │   ├── conf
    │   ├── include
    │   ├── legal
    │   ├── lib
    │   └── release
    ├── Info.plist
    ├── MacOS
    │   └── libjli.dylib
    └── _CodeSignature
        └── CodeResources
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

siraben commented:
Result of `nixpkgs-review pr 157511` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>bluej</li>
    <li>greenfoot</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jetbrains.jdk</li>
  </ul>
</details>

domenkozar commented:
Needs a rebase

siraben commented:
Result of `nixpkgs-review pr 157511` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>bluej</li>
    <li>greenfoot</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>jetbrains.jdk</li>
  </ul>
</details>

felixscheinost committed:
jetbrains.jdk: Fix darwin build

Similarly to how OpenJDK is a source build on Linux but using binaries from Azul the source build of JetbrainsRuntime (OpenJDK fork) is currently broken on Darwin.

I think it makes sense to use binaries for Darwin here as well.

I also updated from 11_0_13-b1751.25 to 11_0_15_1-b2043.56 as we currently have IDE version 2022.1.3 for which https://github.com/JetBrains/JetBrainsRuntime/blob/jbr11/README.md recommends this version.

I tested using `jetbrains.idea-community` which seems to work.

wamserma commented:
This PR is outdated due to https://github.com/NixOS/nixpkgs/pull/183763.
